### PR TITLE
Use es-rollover golang implementation

### PR DESCRIPTION
**Pull request by rubenvp8510**
Signed-off-by: Ruben Vargas <ruben.vp8510@gmail.com>

Updates #3179

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -188,6 +184,10 @@ build-esmapping-generator-linux:
 build-es-index-cleaner:
 	$(GOBUILD) -o ./cmd/es-index-cleaner/es-index-cleaner-$(GOOS)-$(GOARCH) ./cmd/es-index-cleaner/main.go
 
+.PHONY: build-es-rollover
+build-es-rollover:
+	$(GOBUILD) -o ./cmd/es-rollover/es-rollover-$(GOOS)-$(GOARCH) ./cmd/es-rollover/main.go
```
Comment (yurishkuro) [inline Makefile:189]: please add .gitignore in that folder and exclude these new binaries

File: Makefile [hunk 2]
```diff
@@ -109,10 +109,6 @@ grpc-plugin-storage-integration-test:
 	(cd examples/memstore-plugin/ && go build .)
 	STORAGE=grpc-plugin $(MAKE) storage-integration-test
 
-.PHONY: test-compile-es-scripts
-test-compile-es-scripts:
```
Comment (yurishkuro) [inline Makefile:113]: this is called from 
```
./scripts/es-integration-test.sh:97:  make test-compile-es-scripts
```

**Comment by yurishkuro** [comment]
are the CLI flags in the new image compatible with the old python script?

**Comment by pavolloffay** [comment]
The CI didn't pass

File: cmd/es-rollover/app/index_options_test.go [hunk 3]
```diff
@@ -61,7 +61,7 @@ func TestRolloverIndices(t *testing.T) {
 			expected: []expectedValues{
 				{
 					prefix:               "mytenant-jaeger-span-archive",
-					templateName:         "mytenant-jaeger-span",
```
Comment (pavolloffay) [inline cmd/es-rollover/app/index_options_test.go:64]: Why is this changing? Shouldn't the template name keep the prefix name? 

The new golang implementation has to be backwards compatible.

**Review by pavolloffay — APPROVED** [review]

**Comment by pavolloffay** [comment]
> are the CLI flags in the new image compatible with the old python script?

Yes this is the goal

File: cmd/es-rollover/app/index_options_test.go (hunk 3 above)
Comment (rubenvp8510) [reply cmd/es-rollover/app/index_options_test.go:64]: I shouldn't 

https://github.com/jaegertracing/jaeger/pull/3258/files#diff-59567de3853ac61d1a7460823020ee5c2fe7680353df87011f4cd1df1b132f45L83

Look as this calls, the template has no prefix.  It was my mistake to include the prefix in the first PR for this.

Comment (rubenvp8510) [reply cmd/es-rollover/app/index_options_test.go:64]: wait. I see one mistake , don't merge this please.

Comment (rubenvp8510) [reply cmd/es-rollover/app/index_options_test.go:64]: Fixed, you're right, the template name should be prefixed, the mapping name used to build the template should not.

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**